runtests.py

```diff
@@ -311,6 +311,35 @@ def update_cpp11_extension(ext):
 
     return EXCLUDE_EXT
 
+def update_cpp17_extension(ext):
+    """
+        update cpp17 extensions that will run on versions of gcc >=5.0
+    """
+    gcc_version = get_gcc_version(ext.language)
+    if gcc_version:
+        compiler_version = gcc_version.group(1)
+        if float(compiler_version) >= 5.0:
+            ext.extra_compile_args.append("-std=c++17")
+        return ext
+
+    clang_version = get_clang_version(ext.language)
+    if clang_version:
+        ext.extra_compile_args.append("-std=c++17")
+        if sys.platform == "darwin":
+          ext.extra_compile_args.append("-stdlib=libc++")
+          ext.extra_compile_args.append("-mmacosx-version-min=10.13")
+        return ext
+
+    return EXCLUDE_EXT
+
+def require_gcc(version):
+    def check(ext):
+        gcc_version = get_gcc_version(ext.language)
+        if gcc_version:
+            if float(gcc_version.group(1)) >= float(version):
+                return ext
+        return EXCLUDE_EXT
```

Comment:
Shouldn't it be more like "IF we're using gcc/g++, THEN check the version – otherwise, rely on the 'cpp17' tag" ? Excluding everything but g++ here seems too broad – to me. (But I didn't look up the details.)

Reply:
Could I ask which other compilers Cython tests against? I only have access to gcc at the moment, but for the others, I'd have to track down which versions introduced the `<execution>` header.

Reply:
Anything that can be used to compile Python should be allowed: clang, mingw, MSVC, intel, ...

Reply:
Understood, the changes in this PR don't preclude using any of those to compile Cython. However, they do add Cython definitions for a C++17 only feature, which all compilers and all versions support.

Reply:
> Could I ask which other compilers Cython tests against?

The automated tests get run on GCC on Linux, Clang on Mac, and [MSVC on Windows](https://ci.appveyor.com/project/cython/cython).

Putting in version-checks in runtests.py for all known compilers  seems a bit of a pain to me (but there isn't an obvious better way to do it, and there's precedent for c++11). I wonder if it'd be better for the tests to be off by default in runtests.py, and enabled by a command-line argument (which would be turned on in the CIs that Cython controls itself)? (But I'm not confident in this opinion)

Reply:
From what I can see, Clang still does not support this feature (Ctrl-F for "Standardization of Parallelism TS" [here](https://en.cppreference.com/w/cpp/compiler_support#cpp17)), and MSVC supports it as of VS 2017 15.7.

How should we proceed here?

Reply:
I think it's fine to start testing it with one compiler (as long as it's GCC :D ) and add others later, when we feel like it.
This isn't really a feature where I would expect compiler specific breakage being introduced by changes in Cython, and even if we find such problems later, we can also fix them later. And add more compilers at that point, when we hear about one where it's supposed to work but doesn't.